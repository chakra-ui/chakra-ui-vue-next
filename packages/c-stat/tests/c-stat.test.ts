import { TestRenderProps, render, testA11y } from "@chakra-ui/vue-test-utils"
import {
  CStat,
  CStatGroup,
  CStatLabel,
  CStatNumber,
  CStatHelpText,
  CStatArrow,
} from "../src"

const renderComponent = (props: TestRenderProps = {}) => {
  const base = {
    components: {
      CStat,
      CStatGroup,
      CStatLabel,
      CStatNumber,
      CStatHelpText,
      CStatArrow,
    },
    template: `
    <c-stat-group data-testid="group">
      <c-stat>
        <c-stat-label>Sent</c-stat-label>
        <c-stat-number>345,670</c-stat-number>
        <c-stat-help-text>
          <c-stat-arrow type="increase" />
          23.36%
        </c-stat-help-text>
      </c-stat>

      <c-stat>
        <c-stat-label>Clicked</c-stat-label>
        <c-stat-number>45</c-stat-number>
        <c-stat-help-text>
          <c-stat-arrow type="decrease" />
          9.05%
        </c-stat-help-text>
      </c-stat>
    </c-stat-group>
    `,
    ...props,
  }
  return render(base)
}

describe("CStat", () => {
  it("should render properly", () => {
    const { asFragment, getByTestId } = renderComponent()

    expect(asFragment()).toMatchSnapshot()
    expect(getByTestId("group")).toBeTruthy()
  })

  it("should pass a11y tests", async () => {
    await testA11y(renderComponent())
  })
})
