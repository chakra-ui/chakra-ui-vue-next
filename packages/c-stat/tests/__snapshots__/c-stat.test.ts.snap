// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CStat should render properly 1`] = `
<DocumentFragment>
  <div
    class="chakra-stat__group chakra-1mpfoc5"
    data-testid="group"
    role="group"
  >
    <div
      class="chakra-stat chakra-1mbo1ls"
    >
      <dl>
        <dt
          class="chakra-stat__label chakra-w75v0a"
        >
          Sent
        </dt>
        <dd
          class="chakra-stat__number chakra-pz3tqd"
        >
          345,670
        </dd>
        <dd
          class="chakra-stat__help-text chakra-k1lc38"
        >
          <span
            class="chakra-8b45rq"
          >
            increased by
          </span>
          <svg
            aria-hidden="true"
            class="chakra-icon chakra-1enj48r"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12.8,5.4c-0.377-0.504-1.223-0.504-1.6,0l-9,12c-0.228,0.303-0.264,0.708-0.095,1.047 C2.275,18.786,2.621,19,3,19h18c0.379,0,0.725-0.214,0.895-0.553c0.169-0.339,0.133-0.744-0.095-1.047L12.8,5.4z"
              fill="currentColor"
            />
          </svg>
           23.36% 
        </dd>
      </dl>
    </div>
    <div
      class="chakra-stat chakra-1mbo1ls"
    >
      <dl>
        <dt
          class="chakra-stat__label chakra-w75v0a"
        >
          Clicked
        </dt>
        <dd
          class="chakra-stat__number chakra-pz3tqd"
        >
          45
        </dd>
        <dd
          class="chakra-stat__help-text chakra-k1lc38"
        >
          <span
            class="chakra-8b45rq"
          >
            decreased by
          </span>
          <svg
            aria-hidden="true"
            class="chakra-icon chakra-6ohva6"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M21,5H3C2.621,5,2.275,5.214,2.105,5.553C1.937,5.892,1.973,6.297,2.2,6.6l9,12 c0.188,0.252,0.485,0.4,0.8,0.4s0.611-0.148,0.8-0.4l9-12c0.228-0.303,0.264-0.708,0.095-1.047C21.725,5.214,21.379,5,21,5z"
              fill="currentColor"
            />
          </svg>
           9.05% 
        </dd>
      </dl>
    </div>
  </div>
</DocumentFragment>
`;
